internal/pkg/validator/validator.go

```diff
@@ -164,6 +166,22 @@ func (v *wrapper) registerCustomRules() {
 			},
 			ErrorMsg: "{0} does not contain an allowed icon",
 		},
+		Rule{
+			Tag: "mdmax",
+			FuncCtx: func(ctx context.Context, fl validator.FieldLevel) bool {
+				value := fl.Field().String()
+				paramStr := fl.Param()
+				param, err := strconv.Atoi(paramStr)
+				if err != nil {
+					panic(fmt.Sprintf("failed to convert mdmax param `%v` to an int", paramStr))
+				}
+				value = strhelper.StripMarkdown(value)
+				return len(value) <= param
+			},
+			ErrorMsgFunc: func(fe validator.FieldError) string {
+				return fmt.Sprintf("%s exceeded maximum length of %s", fe.Tag(), fe.Param())
```

Comment:
Použil jsem `ErrorMsgFunc` místo `ErrorMsg`, ale moc nechápu, co dělá ten translation v `registerErrorMsg` - je tohle v pořádku?

Reply:
jj je to dobre. `ErrorMsg` je pre staticku chybu ...

Ono to je cele wrapper okolo https://github.com/go-playground/validator.
Ten je super, ma vela pravidiel, ale ma jednu zasadnu chybu, ... a to ze z validacnej funkcie nemozes vratit `error` ale iba `bool`. To obchadzame pomocou `translations` a samostatnej funkcie na generovanie chybovej hlasky. Idealne by to bolo mat vsetko v jednej funkcii, kedze preklady neriesime a vsetko je v EN, ale neda sa ... 

Vid; https://github.com/go-playground/validator/issues/669 ( ma to tag `v11`, aktualna je `v10`, tak sa mozno dockame).

Mozno to casom premigrujeme na inu validacnu libku, ale neviem ci to stoji za ten cas.